client/my-sites/site-settings/form-security.jsx

```diff
@@ -144,9 +159,21 @@ const getFormSettings = ( settings ) =>
 		'sso',
 		'jetpack_sso_match_by_email',
 		'jetpack_sso_require_two_step',
-		'wordpress_api_key',
-	] );
+	];
+
+	if ( includeAkismetKeyField || settings.akismet ) {
+		settingsToPick.push( 'wordpress_api_key' );
+	}
+
+	return pick( settings, settingsToPick );
+};
 
 export default connectComponent(
-	localize( wrapSettingsForm( getFormSettings )( SiteSettingsFormSecurity ) )
+	localize( ( props ) => {
+		const WrappedSiteSettingsFormSecurity = wrapSettingsForm(
+			getFormSettings( props.includeAkismetFields )
+		)( SiteSettingsFormSecurity );
+
+		return <WrappedSiteSettingsFormSecurity { ...props } />;
```

Comment:
I'm afraid this won't work. The inner component will create a new wrapped component on each render, while a regular HOC does it only once, statically. The wrapped component will be losing local state and focus on each rerender.

A better solution would be `getFormSettings` accepting a second parameter (props) so that it can make decisions derived from the props. And the `includeAkismet` value would need to passed from outside somehow, so that `getFormSettings` can read it.

Reply:
Dang, you're right. I'll go the route of giving `getFormSettings` a second arg.

Reply:
Updated!